From 42043c35ac62dcf3b880974b3d0fb762583733ff Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Wed, 27 Nov 2024 12:49:52 +0100
Subject: [PATCH 1/2] profiles/audio: Quiet plug-in warnings

A normal daemon start should not throw warnings on a system in the
default configuration. Quiet the plug-in warnings that require
experimental features to be enabled. They will still appear in the debug
output.

bluetoothd[896]: profiles/audio/micp.c:micp_init() D-Bus experimental not enabled
---
 profiles/audio/media.c | 2 +-
 profiles/audio/micp.c  | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/profiles/audio/media.c b/profiles/audio/media.c
index 1faa1c28969f..efd1d667c270 100644
--- a/profiles/audio/media.c
+++ b/profiles/audio/media.c
@@ -1214,7 +1214,7 @@ static bool endpoint_init_pac(struct media_endpoint *endpoint, uint8_t type,
 	char *name;
 
 	if (!(g_dbus_get_flags() & G_DBUS_FLAG_ENABLE_EXPERIMENTAL)) {
-		warn("D-Bus experimental not enabled");
+		DBG("D-Bus experimental not enabled");
 		*err = -ENOTSUP;
 		return false;
 	}
diff --git a/profiles/audio/micp.c b/profiles/audio/micp.c
index 452027c75da2..3f0845dcb328 100644
--- a/profiles/audio/micp.c
+++ b/profiles/audio/micp.c
@@ -318,7 +318,7 @@ static unsigned int micp_id;
 static int micp_init(void)
 {
 	if (!(g_dbus_get_flags() & G_DBUS_FLAG_ENABLE_EXPERIMENTAL)) {
-		warn("D-Bus experimental not enabled");
+		DBG("D-Bus experimental not enabled");
 		return -ENOTSUP;
 	}
 
-- 
2.48.1


From bca2b4b7d69df7158dc1024c4bf7510ff771f189 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Wed, 27 Nov 2024 12:49:53 +0100
Subject: [PATCH 2/2] plugin: Quiet start-up warnings

It's not an error for a plug-in not to start if the hardware doesn't
support it. Quiet the warnings that require specific hardware.
The messages will still appear in the debug output.

bluetoothd[896]: src/plugin.c:plugin_init() System does not support csip plugin
bluetoothd[896]: src/plugin.c:plugin_init() System does not support micp plugin
bluetoothd[896]: src/plugin.c:plugin_init() System does not support vcp plugin
bluetoothd[896]: src/plugin.c:plugin_init() System does not support mcp plugin
bluetoothd[896]: src/plugin.c:plugin_init() System does not support bass plugin
bluetoothd[896]: src/plugin.c:plugin_init() System does not support bap plugin
---
 src/plugin.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/plugin.c b/src/plugin.c
index 80990f8c3c7e..42ed75ab31f3 100644
--- a/src/plugin.c
+++ b/src/plugin.c
@@ -187,7 +187,7 @@ start:
 		err = plugin->desc->init();
 		if (err < 0) {
 			if (err == -ENOSYS || err == -ENOTSUP)
-				warn("System does not support %s plugin",
+				DBG("System does not support %s plugin",
 							plugin->desc->name);
 			else
 				error("Failed to init %s plugin",
-- 
2.48.1

